src/test/java/com/jcabi/http/wire/LastModifiedCachingWireTest.java

```diff
@@ -0,0 +1,129 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.http.wire;
+
+import com.jcabi.aspects.Tv;
+import com.jcabi.http.Request;
+import com.jcabi.http.mock.MkAnswer;
+import com.jcabi.http.mock.MkContainer;
+import com.jcabi.http.mock.MkGrizzlyContainer;
+import com.jcabi.http.mock.MkQuery;
+import com.jcabi.http.request.JdkRequest;
+import com.jcabi.http.response.RestResponse;
+import java.net.HttpURLConnection;
+import java.util.Collections;
+import java.util.Map;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.hamcrest.core.IsAnything;
+import org.junit.Test;
+
+/**
+ * Test case for {@link LastModifiedCachingWire}.
+ * @author Igor Piddubnyi (igor.piddubnyi@gmail.com)
+ * @version $Id$
+ * @since 1.15
+ */
+public final class LastModifiedCachingWireTest {
+
+    /**
+     * Time long ago.
+     */
+    private static final String LONG_AGO = "Wed, 15 Nov 1995 04:58:08 GMT";
+
+    /**
+     * Test body.
+     * */
+    private static final String BODY = "Test body";
+
+    /**
+     * LastModifiedCachingWire can cache GET requests.
+     * @throws Exception If fails
+     */
+    @Test
+    public void cachesGetRequest() throws Exception {
+        final Map<String, String> headers = Collections.singletonMap(
+            LastModifiedCachingWire.LAST_MODIFIED,
+            LastModifiedCachingWireTest.LONG_AGO
+        );
+        final MkContainer container = new MkGrizzlyContainer()
+            .next(
+                new MkAnswer.Simple(
+                    HttpURLConnection.HTTP_OK,
+                    headers.entrySet(),
+                    BODY.getBytes()
+                )
+            )
+            .next(
+                new MkAnswer.Simple(HttpURLConnection.HTTP_NOT_MODIFIED),
+                new IsAnything<MkQuery>(),
+                Tv.TEN
+            ).start();
+        try {
+            final Request req = new JdkRequest(container.home())
+                .through(LastModifiedCachingWire.class);
+            for (int idx = 0; idx < Tv.TEN; ++idx) {
+                req.fetch().as(RestResponse.class)
+                    .assertStatus(HttpURLConnection.HTTP_OK)
+                    .assertBody(Matchers.equalTo(BODY));
+            }
+            MatcherAssert.assertThat(
+                container.queries(), Matchers.equalTo(Tv.TEN)
+            );
+        } finally {
+            container.stop();
+        }
+    }
+
+    /**
+     * LastModifiedCachingWire is skips cache for PUT requests and always send
+     * request to original destination.
+     * @throws Exception If something goes wrong inside
+     */
+    @Test
+    public void ignoresPutRequest() throws Exception {
```

Comment:
@cvrebert you are right but I don't see the possibility to test that.
The only test is possible - to configure MkGrizzlyContainer to reply only to requests without If-Modified-Since header with response that contains Last-Modified header. But real server will never reply on PUT with Last-Modified header.